xos.h

```diff
@@ -101,36 +102,44 @@ namespace xos
     int lookahead_;
   };
 
-} // namespace xos
+  class StrAST
+  {
+  public:
+    std::string StrVal;
 
-class StrAST
-{
-  std::string StrVal;
+    StrAST(const std::string &StrVal) : StrVal(StrVal) {}
+  };
 
-public:
-  StrAST(const std::string &StrVal) : StrVal(StrVal) {}
-};
+  class PrototypeAST
+  {
+  public:
+    std::string Name;
+    std::vector<std::string> Args;
+    std::string returnType;
+    PrototypeAST(const std::string &Name, std::vector<std::string> Args, std::string returnType)
```

Comment:
We should also have a `Type` node. The args should be either two vectors (or a vector of pairs): one vector denoting the name and one denoting the `Type`.

When passing a vector, also pass it as a `const` reference to avoid copying the vector. Same with a string.